### ISSUE TITLE
Clang 9.0.1 cannot compile the test application

### ISSUE THREAD

**Issue by lahwaacz** [Lek-sys/LeksysINI#5]
```
$ clang++ test_app.cpp 
In file included from test_app.cpp:2:
./iniparser.hpp:796:50: error: no viable conversion from 'std::map<INI::Value, INI::Value, std::less<INI::Value>, std::allocator<std::pair<const INI::Value, INI::Value> > >::const_iterator' (aka '_Rb_tree_const_iterator<std::pair<const INI::Value, INI::Value> >') to 'std::map<Value, Value>::iterator' (aka '_Rb_tree_iterator<std::pair<const INI::Value, INI::Value> >')
            for (std::map<Value,Value>::iterator it = _val->begin(); it != _val->end(); ++it)
                                                 ^    ~~~~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../include/c++/9.2.0/bits/stl_tree.h:256:12: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::map<INI::Value, INI::Value, std::less<INI::Value>, std::allocator<std::pair<const INI::Value, INI::Value> > >::const_iterator' (aka '_Rb_tree_const_iterator<std::pair<const INI::Value, INI::Value> >') to 'const std::_Rb_tree_iterator<std::pair<const INI::Value, INI::Value> > &' for 1st argument
    struct _Rb_tree_iterator
           ^
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../include/c++/9.2.0/bits/stl_tree.h:256:12: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'std::map<INI::Value, INI::Value, std::less<INI::Value>, std::allocator<std::pair<const INI::Value, INI::Value> > >::const_iterator' (aka '_Rb_tree_const_iterator<std::pair<const INI::Value, INI::Value> >') to 'std::_Rb_tree_iterator<std::pair<const INI::Value, INI::Value> > &&' for 1st argument
1 error generated.
```

**Comment by lahwaacz** [comment]
Simple workaround for C++11 is to write
```cpp
for (auto it = _val->begin(); it != _val->end(); ++it)
```
instead of
```cpp
for (std::map<Value,Value>::iterator it = _val->begin(); it != _val->end(); ++it)
```

**Comment by tagorereddy** [comment]
This should fix things #8

**Comment by Lek-sys** [comment]
Fixed! Thanks to @tagorereddy
